### ISSUE TITLE
SET @@mydb_head = HASHOF() panics when session variables is spelled wrong

### ISSUE THREAD

**Issue by VinaiRachakonda** [dolthub/dolt#1345]
```
junk> SET @@mjunk_head = HASHOF("master");
panic: unresolved function is a placeholder node, but Type was called

goroutine 1 [running]:
github.com/dolthub/go-mysql-server/sql/expression.(*UnresolvedFunction).Type(0xc0005668d0, 0x3456b80, 0x3456b80)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/expression/unresolved.go:126 +0x39
github.com/dolthub/go-mysql-server/sql/analyzer.resolveSetVariables.func1(0x29103c0, 0xc000568600, 0x2, 0x2, 0x29103c0, 0xc000568600)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/analyzer/resolve_variables.go:68 +0x538
github.com/dolthub/go-mysql-server/sql/expression.TransformUp(0x29103c0, 0xc000568520, 0xc00099eba8, 0xc000181aa0, 0xc000568540, 0x24a7a01, 0x2554900)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/expression/transform.go:47 +0x1a3
github.com/dolthub/go-mysql-server/sql/plan.TransformExpressions(0x290b5c0, 0xc000568540, 0xc00099eba8, 0x0, 0x0, 0x28d4900, 0xc00099eb60)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/plan/transform.go:198 +0xf7
github.com/dolthub/go-mysql-server/sql/plan.TransformExpressionsUp(0x290b5c0, 0xc000568540, 0xc00099eba8, 0xc00099ebb8, 0x100d45b, 0x10, 0x2347260)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/plan/transform.go:163 +0x27d
github.com/dolthub/go-mysql-server/sql/analyzer.resolveSetVariables(0xc0005643f0, 0xc0000a4140, 0x290b5c0, 0xc000568540, 0x0, 0x1, 0xc000568540, 0x0, 0x0)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/analyzer/resolve_variables.go:37 +0xb8
github.com/dolthub/go-mysql-server/sql/analyzer.(*Batch).evalOnce(0xc00049a1e0, 0xc0005643f0, 0xc0000a4140, 0x290b5c0, 0xc000568540, 0x0, 0x290b5c0, 0xc000568540, 0x0, 0x0)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/analyzer/batch.go:93 +0x1c5
github.com/dolthub/go-mysql-server/sql/analyzer.(*Batch).Eval(0xc00049a1e0, 0xc0005643f0, 0xc0000a4140, 0x290b5c0, 0xc000568540, 0x0, 0x290b5c0, 0xc000568540, 0x0, 0x0)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/analyzer/batch.go:54 +0xe5
github.com/dolthub/go-mysql-server/sql/analyzer.(*Analyzer).Analyze(0xc0000a4140, 0xc0005643f0, 0x290b5c0, 0xc000568540, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/sql/analyzer/analyzer.go:346 +0x28c
github.com/dolthub/go-mysql-server.(*Engine).QueryWithBindings(0xc00049a390, 0xc0005642d0, 0xc0000485d0, 0x24, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/engine.go:202 +0x2d2
github.com/dolthub/go-mysql-server.(*Engine).Query(...)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/go-mysql-server@v0.8.1-0.20210217184311-572460796d05/engine.go:151
github.com/dolthub/dolt/go/cmd/dolt/commands.(*sqlEngine).query(0xc00049e2a0, 0xc000549c20, 0xc0000485d0, 0x24, 0x0, 0x0, 0x4, 0x4, 0xc000562170, 0x3, ...)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/commands/sql.go:1349 +0x65
github.com/dolthub/dolt/go/cmd/dolt/commands.processQuery(0xc000549c20, 0xc0000485d0, 0x24, 0xc00049e2a0, 0x0, 0x0, 0xffffffffffffffff, 0xc0000485d0, 0x24, 0xc00099f1e0, ...)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/commands/sql.go:852 +0x114
github.com/dolthub/dolt/go/cmd/dolt/commands.runShell.func3(0xc0003a0270)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/commands/sql.go:668 +0x2c7
github.com/dolthub/ishell.handleUninterpretedInput(0xc0001ac120, 0xc0000485d0, 0x24, 0x0, 0xc000180f01)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/ishell@v0.0.0-20210205014355-16a4ce758446/ishell.go:282 +0x146
github.com/dolthub/ishell.(*Shell).run(0xc0001ac120)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/ishell@v0.0.0-20210205014355-16a4ce758446/ishell.go:242 +0x245
github.com/dolthub/ishell.(*Shell).Run(0xc0001ac120)
	/Users/vinairachakonda/go/pkg/mod/github.com/dolthub/ishell@v0.0.0-20210205014355-16a4ce758446/ishell.go:136 +0x39
github.com/dolthub/dolt/go/cmd/dolt/commands.runShell(0xc000549c20, 0xc00049e2a0, 0xc0008a5ec0, 0xc00039e000, 0x0, 0xc0000a4050)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/commands/sql.go:691 +0x6be
github.com/dolthub/dolt/go/cmd/dolt/commands.execShell(0xc000549c20, 0x4200100, 0xc0008a5ec0, 0xc00039e000, 0x0, 0x2916a00, 0xc00018b6bc)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/commands/sql.go:329 +0xe5
github.com/dolthub/dolt/go/cmd/dolt/commands.SqlCmd.Exec(0x265700f, 0x6, 0x28ff580, 0xc0005a2780, 0xc00059e418, 0x8, 0xc000198030, 0x0, 0x0, 0xc000520fd0, ...)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/commands/sql.go:311 +0x1697
github.com/dolthub/dolt/go/cmd/dolt/cli.SubCommandHandler.Exec(0x26553b4, 0x4, 0x266a3ae, 0x11, 0xc0003fb200, 0x24, 0x24, 0x0, 0x28ff580, 0xc0005a2780, ...)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/cli/command.go:165 +0x4bd
main.runMain(0x0)
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/dolt.go:265 +0x1049
main.main()
	/Users/vinairachakonda/go/src/dolthub/dolt/go/cmd/dolt/dolt.go:111 +0x25
```
My db was named junk but I accidentally types @@mjunk_head. That caused the panic.

**Comment by VinaiRachakonda** [comment]
This bug doesn't exist anymore. Not sure which pr fixed it though

**Comment by mango-lzp** [comment]
error on line 1 for query set @@cloud_head = HASHOF('master'): Variable 'cloud_head' is a read only variable.
how to fix this, I just upgrade dolt version from 0.2.x to 0.5.x
